cub/cub/device/dispatch/kernels/transform.cuh

```diff
@@ -125,6 +128,125 @@ _CCCL_DEVICE void transform_kernel_impl(
   }
 }
 
+template <typename T, int LoadLength>
+using vector_type = ::cuda::std::_If<
```

Comment:
Nitpick: We should rather use `::cuda::std::conditional_t` here `::cuda::std::_If` is something I would love to get riid of

Suggestion: This would better be a `_CCCL_CONSTEVAL` function with a sequence of `if constexpr` statements